Installation issue: libevent

### Steps to reproduce the issue

```console
$ spack spec -I libevent
Input spec
--------------------------------
 -   libevent

Concretized
--------------------------------
 -   libevent@2.1.12%gcc@13.2.1+openssl build_system=autotools arch=linux-fedora38-zen2
[+]      ^autoconf@2.69%gcc@13.2.1 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-fedora38-zen2
[+]          ^m4@1.4.19%gcc@13.2.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-fedora38-zen2
[+]              ^diffutils@3.9%gcc@13.2.1 build_system=autotools arch=linux-fedora38-zen2
[+]                  ^libiconv@1.17%gcc@13.2.1 build_system=autotools libs=shared,static arch=linux-fedora38-zen2
[+]              ^libsigsegv@2.14%gcc@13.2.1 build_system=autotools arch=linux-fedora38-zen2
[+]          ^perl@5.38.0%gcc@13.2.1+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-fedora38-zen2
[+]              ^berkeley-db@18.1.40%gcc@13.2.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-fedora38-zen2
[+]              ^bzip2@1.0.8%gcc@13.2.1~debug~pic+shared build_system=generic arch=linux-fedora38-zen2
[+]              ^gdbm@1.23%gcc@13.2.1 build_system=autotools arch=linux-fedora38-zen2
[+]                  ^readline@8.2%gcc@13.2.1 build_system=autotools patches=bbf97f1 arch=linux-fedora38-zen2
[+]                      ^ncurses@6.4%gcc@13.2.1~symlinks+termlib abi=none build_system=autotools arch=linux-fedora38-zen2
[+]                          ^pkgconf@1.9.5%gcc@13.2.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^automake@1.16.5%gcc@13.2.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^gmake@4.4.1%gcc@13.2.1~guile build_system=generic arch=linux-fedora38-zen2
[+]      ^libtool@2.4.7%gcc@13.2.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^openssl@3.1.3%gcc@13.2.1~docs+shared build_system=generic certs=mozilla arch=linux-fedora38-zen2
[+]          ^ca-certificates-mozilla@2023-05-30%gcc@13.2.1 build_system=generic arch=linux-fedora38-zen2
[+]          ^zlib-ng@2.1.4%gcc@13.2.1+compat+opt build_system=autotools arch=linux-fedora38-zen2
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Installing libevent-2.1.12-r2q53e4tiqwcxjmtagvi4bsftgovtp35 [15/15]
==> No binary for libevent-2.1.12-r2q53e4tiqwcxjmtagvi4bsftgovtp35 found: installing from source
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/92/92e6de1be9ec176428fd2367677e61ceffc2ee1cb119035037a27d346b0403bb.tar.gz
==> Ran patch() for libevent
==> libevent: Executing phase: 'autoreconf'
==> libevent: Executing phase: 'configure'
==> libevent: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

18 errors found in build log:
     333       true; \
     334    else \
     335       echo "No Python installed; stubbing out RPC test." >&2; \
     336       echo " "> test/regress.gen.c; \
     337       echo "#define NO_PYTHON_EXISTS" > test/regress.gen.h; \
     338    fi
  >> 339    /usr/bin/env: 'python': No such file or directory
     340    event_rpcgen.py failed, ./test/regress.gen.\[ch\] will be reused.
     341    rm -f stamp-h1
     342    touch config.h.in
     343    cd . && /bin/sh ./config.status config.h
     344    config.status: creating config.h
     345    config.status: config.h is unchanged

     ...

     401    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     402    libtool: and run autoconf again.
     403    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     404    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     405    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     406    libtool: and run autoconf again.
  >> 407    make[1]: *** [Makefile:1865: buffer.lo] Error 63
     408    make[1]: *** Waiting for unfinished jobs....
     409    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     410    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     411    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     412    libtool: and run autoconf again.
     413    libtool: Version mismatch error.  This is libtool 2.4.7, but the

     ...

     415    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     416    libtool: and run autoconf again.
     417    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     418    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     419    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     420    libtool: and run autoconf again.
  >> 421    make[1]: *** [Makefile:1865: bufferevent_pair.lo] Error 63
     422    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     423    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     424    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     425    libtool: and run autoconf again.
     426    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     427    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     428    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     429    libtool: and run autoconf again.
  >> 430    make[1]: *** [Makefile:1865: event.lo] Error 63
  >> 431    make[1]: *** [Makefile:1865: evmap.lo] Error 63
     432    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     433    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     434    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     435    libtool: and run autoconf again.
  >> 436    make[1]: *** [Makefile:1865: bufferevent.lo] Error 63
  >> 437    make[1]: *** [Makefile:1865: evutil.lo] Error 63
     438    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     439    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     440    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     441    libtool: and run autoconf again.
  >> 442    make[1]: *** [Makefile:1865: bufferevent_filter.lo] Error 63
  >> 443    make[1]: *** [Makefile:1865: bufferevent_sock.lo] Error 63
  >> 444    make[1]: *** [Makefile:1865: bufferevent_ratelim.lo] Error 63
     445    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     446    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     447    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     448    libtool: and run autoconf again.
     449    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     450    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.

     ...

     463    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     464    libtool: and run autoconf again.
     465    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     466    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     467    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     468    libtool: and run autoconf again.
  >> 469    make[1]: *** [Makefile:1865: select.lo] Error 63
  >> 470    make[1]: *** [Makefile:1865: evthread.lo] Error 63
  >> 471    make[1]: *** [Makefile:1865: evutil_time.lo] Error 63
  >> 472    make[1]: *** [Makefile:1865: listener.lo] Error 63
  >> 473    make[1]: *** [Makefile:1865: evutil_rand.lo] Error 63
  >> 474    make[1]: *** [Makefile:1865: log.lo] Error 63
     475    libtool: Version mismatch error.  This is libtool 2.4.7, but the
     476    libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
     477    libtool: You should recreate aclocal.m4 with macros from libtool 2.4.7
     478    libtool: and run autoconf again.
  >> 479    make[1]: *** [Makefile:1865: strlcpy.lo] Error 63
     480    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-libevent-2.1.12-r2q53e4tiqwcxjmtagvi4bsftgovtp35/spack-src'
  >> 481    make: *** [Makefile:1389: all] Error 2

</pre></details>


### Information on your system

* **Spack:** 0.21.0 (65d3221a9c436e77af9e802e543c71bd702ff4e2)
* **Python:** 3.11.6
* **Platform:** linux-fedora38-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/13342952/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/13342953/spack-build-env.txt)
[config.log](https://github.com/spack/spack/files/13342954/config.log)



### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/libevent/package.py
+++ b/var/spack/repos/builtin/packages/libevent/package.py
@@ -59,7 +59,7 @@ class Libevent(AutotoolsPackage):
         return LibraryList(libs)
 
     def autoreconf(self, spec, prefix):
-        Executable("./autogen.sh")()
+        autoreconf("--force", "--install", "--symlink")
 
     def configure_args(self):
         spec = self.spec

